qos/evm/endpoint_selection.go

```diff
@@ -30,7 +30,7 @@ func (ss *serviceState) Select(availableEndpoints []protocol.EndpointAddr) (prot
 	}
 
 	if len(filteredEndpointsAddr) == 0 {
-		logger.Error().Msg("all endpoints failed validation; selecting a random endpoint.")
+		logger.Warn().Msg("SELECTING A RANDOM ENDPOINT because all endpoints failed validation.")
```

Comment:
IMO this should be a `Error`.

My rationale for making it a `.Error` is that when debugging this issue (ie during my QoS investigation) I generally am limiting logs to `"error"` level only to reduce the excessive log output and this is one log that I frequently need to find in this situation.